etcdserver/v3_server.go

```diff
@@ -802,3 +802,14 @@ func (s *EtcdServer) linearizableReadNotify(ctx context.Context) error {
 		return ErrStopped
 	}
 }
+
+func (s *EtcdServer) AuthInfoFromCtx(ctx context.Context) (*auth.AuthInfo, error) {
+	if s.Cfg.ClientCertAuthEnabled {
```

Comment:
admin can disable cert auth. If we do not check here we might make mistake, like now command name "foo" == user "foo".